lib/view_component/slot.rb

```diff
@@ -2,6 +2,28 @@
 
 module ViewComponent
   class Slot
-    attr_accessor :content
+    attr_accessor :content, :parent
+    attr_writer :_component_instance
+
+    # Parent must be `nil` for v1
+    def initialize(parent = nil)
+      @parent = parent
+    end
+
+    def to_s
+      if defined?(@_component_instance)
+        parent.render(@_component_instance) { content }
+      else
+        content
+      end
+    end
+
+    def method_missing(symbol, *args, &block)
+      if defined?(@_component_instance)
+        @_component_instance.public_send(symbol, *args, &block)
+      else
+        parent.send(symbol, *args, &block)
```

Comment:
Let's leave this out, so sub-components are required to explicitly reference the parent.